File: lokerpc/codegen/testdata/optional-refs.json.go
package optionalrefs

import (
	"context"
	"time"

	"github.com/LOKE/pkg/lokerpc"
)

type Count int32

type CountAlias Count

type CreatedAt time.Time

type Filter struct {
	Count       *Count      `json:"count,omitempty"`
	CountAlias  *CountAlias `json:"countAlias,omitempty"`
	CreatedAt   CreatedAt   `json:"createdAt,omitzero"`
	InlineCount *int32      `json:"inlineCount,omitempty"`
	MaybeCount  MaybeCount  `json:"maybeCount,omitempty"`
	Ratio       Ratio       `json:"ratio,omitempty"`
	Tags        Tags        `json:"tags,omitzero"`
}

type MaybeCount *int32

type Ratio float64

type Tags []string

type SearchResponse []Count

type OptionalRefsService interface {
	Search(context.Context, Filter) (*SearchResponse, error)
}

type OptionalRefsRPCClient struct {
	lokerpc.Client
}

func (c OptionalRefsRPCClient) Search(ctx context.Context, req Filter) (*SearchResponse, error) {
	var res SearchResponse
	err := c.DoRequest(ctx, "search", req, &res)
	if err != nil {
		return nil, err
	}
	return &res, nil
}

File: lokerpc/codegen/testdata/optional-refs.json.ts
import { RPCContextClient } from "@loke/http-rpc-client";
import { Context } from "@loke/context";

export type Coords = {
  lat: number;
  lng: number;
};

export type Count = number;

export type CountAlias = Count;

export type CreatedAt = string;

export type CreatedAtAlias = CreatedAt;

export type Filter = {
  coords?: Coords;
  count?: Count;
  countAlias?: CountAlias;
  createdAt?: CreatedAt;
  createdAtAlias?: CreatedAtAlias;
  inlineCount?: number;
  maybeCount?: MaybeCount;
  ratio?: Ratio;
  tags?: Tags;
};

export type MaybeCount = number | null;

export type Ratio = number;

export type Tags = string[];

export type SearchResponse = Count[];

/**
 * 
 */
export class OptionalRefsService extends RPCContextClient {
  constructor(baseUrl: string) {
    super(baseUrl, "optional-refs")
  }
  /**
   * search method
   */
  search(ctx: Context, req: Filter): Promise<SearchResponse> {
    return this.request(ctx, "search", req);
  }
}
